// @pcd/pcd-types/src/pcd.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isPCDArgument = exports.isObjectArgument = exports.isBooleanArgument = exports.isBigIntArgument = exports.isNumberArgument = exports.isStringArgument = exports.ArgumentTypeName = void 0;
var ArgumentTypeName;
(function (ArgumentTypeName) {
    ArgumentTypeName["String"] = "String";
    ArgumentTypeName["Number"] = "Number";
    ArgumentTypeName["BigInt"] = "BigInt";
    ArgumentTypeName["Boolean"] = "Boolean";
    ArgumentTypeName["Object"] = "Object";
    ArgumentTypeName["PCD"] = "PCD";
    ArgumentTypeName["Unknown"] = "Unknown";
})(ArgumentTypeName = exports.ArgumentTypeName || (exports.ArgumentTypeName = {}));
function isStringArgument(arg) {
    return arg.argumentType === ArgumentTypeName.String;
}
exports.isStringArgument = isStringArgument;
function isNumberArgument(arg) {
    return arg.argumentType === ArgumentTypeName.Number;
}
exports.isNumberArgument = isNumberArgument;
function isBigIntArgument(arg) {
    return arg.argumentType === ArgumentTypeName.BigInt;
}
exports.isBigIntArgument = isBigIntArgument;
function isBooleanArgument(arg) {
    return arg.argumentType === ArgumentTypeName.Boolean;
}
exports.isBooleanArgument = isBooleanArgument;
function isObjectArgument(arg) {
    return arg.argumentType === ArgumentTypeName.Object;
}
exports.isObjectArgument = isObjectArgument;
function isPCDArgument(arg) {
    return arg.argumentType === ArgumentTypeName.PCD;
}
exports.isPCDArgument = isPCDArgument;
