// zuauth/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ZUPASS_URL: () => ZUPASS_URL,
  isZupassPublicKey: () => isZupassPublicKey,
  openZKEdDSAEventTicketPopup: () => openZKEdDSAEventTicketPopup,
  openZupassPopup: () => openZupassPopup,
  parseSerializedPCD: () => parseSerializedPCD,
  supportedEvents: () => supportedEvents,
  supportedProducs: () => supportedProducs,
  usePopup: () => usePopup,
  useZuAuth: () => useZuAuth,
  useZupassPopupMessages: () => useZupassPopupMessages,
  zupassPublicKey: () => zupassPublicKey
});
module.exports = __toCommonJS(src_exports);

// src/passport-interface/PassportPopup.ts
var import_react = require("react");
function useZupassPopupMessages() {
  const [pcdStr, setPCDStr] = (0, import_react.useState)("");
  const [pendingPCDStr, setPendingPCDStr] = (0, import_react.useState)("");
  (0, import_react.useEffect)(() => {
    function receiveMessage(ev) {
      if (ev.data.encodedPCD) {
        console.log("Received PCD", ev.data.encodedPCD);
        setPCDStr(ev.data.encodedPCD);
      } else if (ev.data.encodedPendingPCD) {
        console.log(ev.data);
        setPendingPCDStr(ev.data.encodedPendingPCD);
      }
    }
    window.addEventListener("message", receiveMessage, false);
    return () => window.removeEventListener("message", receiveMessage);
  }, []);
  return [pcdStr, pendingPCDStr];
}
function usePopup() {
  const [error, setError] = (0, import_react.useState)("");
  (0, import_react.useEffect)(() => {
    if (window.opener == null) {
      setError("Not a popup window");
      return;
    }
    let params;
    if (window.location.href.includes(window.location.origin + "/#/")) {
      const url = new URL(window.location.href.replace("#", ""));
      params = url.searchParams;
    } else {
      params = new URLSearchParams(window.location.search);
    }
    const paramsProofUrl = params.get("proofUrl");
    const paramsProof = params.get("proof");
    const paramsEncodingPendingPCD = params.get("encodedPendingPCD");
    const finished = params.get("finished");
    if (paramsProofUrl != null) {
      window.location.href = paramsProofUrl;
    } else if (finished) {
      if (paramsProof != null) {
        window.opener.postMessage({ encodedPCD: paramsProof }, "*");
      }
      window.close();
      setTimeout(() => {
        setError("Finished. Please close this window.");
      }, 1e3 * 3);
    } else if (paramsEncodingPendingPCD != null) {
      window.opener.postMessage({ encodedPendingPCD: paramsEncodingPendingPCD }, "*");
      window.close();
      setTimeout(() => {
        setError("Finished. Please close this window.");
      }, 1e3 * 3);
    }
  }, []);
  return error;
}
function openZupassPopup(popupUrl, proofUrl) {
  const url = `${popupUrl}?proofUrl=${encodeURIComponent(proofUrl)}`;
  window.open(url, "_blank", "width=450,height=600,top=100,popup");
}

// src/utils.ts
var import_eddsa_ticket_pcd = require("@pcd/eddsa-ticket-pcd");
var import_pcd_types = require("@pcd/pcd-types");
var import_semaphore_identity_pcd = require("@pcd/semaphore-identity-pcd");
var import_zk_eddsa_event_ticket_pcd = require("@pcd/zk-eddsa-event-ticket-pcd");

// src/passport-interface/PassportInterface.ts
function constructZupassPcdGetRequestUrl(zupassClientUrl, returnUrl, pcdType, args, options) {
  const req = {
    type: "Get" /* Get */,
    returnUrl,
    args,
    pcdType,
    options
  };
  const encReq = encodeURIComponent(JSON.stringify(req));
  return `${zupassClientUrl}#/prove?request=${encReq}`;
}

// src/config.ts
var ZUPASS_URL = "https://zupass.org";
var zupassPublicKey = [
  "05e0c4e8517758da3a26c80310ff2fe65b9f85d89dfc9c80e6d0b6477f88173e",
  "29ae64b615383a0ebb1bc37b3a642d82d37545f0f5b1444330300e4c4eedba3f"
];
var ticketTypes = {
  ZuzaluResident: [
    {
      eventId: "5de90d09-22db-40ca-b3ae-d934573def8b",
      productId: "5ba4cd9e-893c-4a4a-b15b-cf36ceda1938"
    }
  ],
  ZuzaluOrganizer: [
    {
      eventId: "5de90d09-22db-40ca-b3ae-d934573def8b",
      productId: "10016d35-40df-4033-a171-7d661ebaccaa"
    }
  ],
  ZuzaluVisitor: [
    {
      eventId: "5de90d09-22db-40ca-b3ae-d934573def8b",
      productId: "53b518ed-e427-4a23-bf36-a6e1e2764256"
    }
  ],
  ZuConnectResident: [
    {
      eventId: "91312aa1-5f74-4264-bdeb-f4a3ddb8670c",
      productId: "cc9e3650-c29b-4629-b275-6b34fc70b2f9"
    },
    {
      eventId: "54863995-10c4-46e4-9342-75e48b68d307",
      productId: "d2123bf9-c027-4851-b52c-d8b73fc3f5af"
    },
    {
      eventId: "797de414-2aec-4ef8-8655-09df7e2b6cc6",
      productId: "d3620f38-56a9-4235-bea8-0d1dba6bb623"
    },
    {
      eventId: "a6109324-7ca0-4198-9583-77962d1b9d53",
      productId: "a6109324-7ca0-4198-9583-77962d1b9d53"
    }
  ]
};
var supportedEvents = Object.values(ticketTypes).flatMap((items) => items.map((item) => item.eventId));
var supportedProducs = Object.values(ticketTypes).flatMap((items) => items.map((item) => item.productId));

// src/utils.ts
function openZKEdDSAEventTicketPopup(fieldsToReveal, watermark, externalNullifier, validEventIds = supportedEvents, validProductIds = supportedProducs, popupRoute = "popup", proofTitle = "ZKEdDSA Ticket Proof", proofDescription = "ZKEdDSA Ticket PCD Request") {
  const args = {
    ticket: {
      argumentType: import_pcd_types.ArgumentTypeName.PCD,
      pcdType: import_eddsa_ticket_pcd.EdDSATicketPCDPackage.name,
      value: void 0,
      userProvided: true,
      validatorParams: {
        eventIds: validEventIds,
        productIds: validProductIds,
        notFoundMessage: "No eligible PCDs found"
      }
    },
    identity: {
      argumentType: import_pcd_types.ArgumentTypeName.PCD,
      pcdType: import_semaphore_identity_pcd.SemaphoreIdentityPCDPackage.name,
      value: void 0,
      userProvided: true
    },
    validEventIds: {
      argumentType: import_pcd_types.ArgumentTypeName.StringArray,
      value: validEventIds.length != 0 ? validEventIds : void 0,
      userProvided: false
    },
    fieldsToReveal: {
      argumentType: import_pcd_types.ArgumentTypeName.ToggleList,
      value: fieldsToReveal,
      userProvided: false
    },
    watermark: {
      argumentType: import_pcd_types.ArgumentTypeName.BigInt,
      value: watermark.toString(),
      userProvided: false
    },
    externalNullifier: {
      argumentType: import_pcd_types.ArgumentTypeName.BigInt,
      value: externalNullifier ? externalNullifier.toString() : watermark.toString(),
      userProvided: false
    }
  };
  const popupUrl = window.location.origin + "/" + popupRoute;
  const proofUrl = constructZupassPcdGetRequestUrl(
    ZUPASS_URL,
    popupUrl,
    import_zk_eddsa_event_ticket_pcd.ZKEdDSAEventTicketPCDPackage.name,
    args,
    {
      genericProveScreen: true,
      title: proofTitle,
      description: proofDescription
    }
  );
  openZupassPopup(popupUrl, proofUrl);
}
function parseSerializedPCD(serializedPCD) {
  try {
    const { pcd } = JSON.parse(serializedPCD);
    return pcd;
  } catch (e) {
    return void 0;
  }
}
function isZupassPublicKey(publicKey) {
  return zupassPublicKey[0] === publicKey[0] && zupassPublicKey[1] === publicKey[1];
}

// src/hooks/useZuAuth.ts
function useZuAuth() {
  const [serializedPCD] = useZupassPopupMessages();
  return { authenticate: openZKEdDSAEventTicketPopup, pcd: parseSerializedPCD(serializedPCD) };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ZUPASS_URL,
  isZupassPublicKey,
  openZKEdDSAEventTicketPopup,
  openZupassPopup,
  parseSerializedPCD,
  supportedEvents,
  supportedProducs,
  usePopup,
  useZuAuth,
  useZupassPopupMessages,
  zupassPublicKey
});
